lib/Target/AVR/AVRInstrInfo.cpp

```diff
@@ -45,6 +45,30 @@ extern bool RA_InSpillerCode;
 AVRInstrInfo::AVRInstrInfo()
     : AVRGenInstrInfo(AVR::ADJCALLSTACKDOWN, AVR::ADJCALLSTACKUP), RI() {}
 
+// FIXME: BEFORE MERGING: Is there a cleaner tablegen method of doing this?
```

Comment:
I think I remembered the [calling convention notes](https://gcc.gnu.org/wiki/avr-gcc) saying that `r0` should generally be `0` but it doesn't look that way.

IIRC @agnat implemented clearing `r0` after multiplication.

Did you figure out the weird `mov`s?